### PR TITLE
Feat/deliverable1

### PR DESCRIPTION
**Pull request by johannrp27**
1.  Create the UI layout for your app, it should include the following elements:
    1.1. A Header containing your Ecommerce Store logo/name, a search input, and a shopping cart icon. For now, all the elements in the header should be disabled, you only need to create the corresponding UI elements without the functionality.
    1.2. The Content view corresponding to the section that will be rendered (in this case, the Home Page)
    1.3. A Footer containing the message “Ecommerce created during Wizeline’s Academy React Bootcamp”

2. Create the Home Page including the following blocks:
    2.1. A Slider to display the featured banners from [this mock file](https://github.com/wizelineacademy/react-bootcamp-capstone-4/blob/main/mocks/en-us/featured-banners.json)
    2.2. A Carousel or Grid of the product categories from [this mock file](https://github.com/wizelineacademy/react-bootcamp-capstone-4/blob/main/mocks/en-us/product-categories.json)
    2.3. A Grid of Featured Products from [this mock file](https://github.com/wizelineacademy/react-bootcamp-capstone-4/blob/main/mocks/en-us/featured-products.json). For each element on this grid, you should show at least the main image of the product, its name, category, and price.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,80 +2,29 @@
...
   "version": "0.1.0",
   "private": true,
+  "scripts": {
+    "start": "react-scripts start",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
   "dependencies": {
+    "@fortawesome/fontawesome-svg-core": "^6.1.1",
```
Comment (buty619) [inline package.json:12]: i don't think is a good idea install this huge library just to use some few icons, i prefer if you use it as svg react component or as image whit css.

File: src/App.jsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import Home from './views/Home';
+import { useFeaturedBanners } from './utils/hooks/useFeaturedBanners';
+
+function App() {
+  const { data, isLoading } = useFeaturedBanners();
+  console.log(data, isLoading);
```
Comment (buty619) [inline src/App.jsx]: please remove this console.log

File: src/components/BannerSlider.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import "react-responsive-carousel/lib/styles/carousel.min.css" // requires a loader
+import { Carousel } from 'react-responsive-carousel'
+import PropTypes from 'prop-types';
+import styles from '../styles/BannerSlider.module.scss'
+
+const BannerSlider = ({data}) => {
+  const banners = data.results;
```
Comment (buty619) [inline src/components/BannerSlider.jsx]: you can destructure this in props as `const BannerSlider = ({data: results})`

File: src/react-fontawesome-config.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { library } from "@fortawesome/fontawesome-svg-core";
+
+import { faCartShopping, faMagnifyingGlass } from "@fortawesome/free-solid-svg-icons";
+
+import { } from "@fortawesome/free-regular-svg-icons";
```
Comment (buty619) [inline src/react-fontawesome-config.js:5]: why is this need it? i think it's better not use this library.

File: src/views/GridCategories.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import CardCategory from '../components/CardCategory';
+import styles from '../styles/GridCategories.module.scss';
+
+const GridCategories = ({data}) => {
```
Comment (buty619) [inline src/views/GridCategories.jsx]: same comment as above.

File: src/views/GridFeatured.jsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import PropTypes from 'prop-types';
+import styles from '../styles/GridFeatured.module.scss';
+import CardFeatured from '../components/CardFeatured';
+
+const GridFeatured = ({data}) => {
+  const featuredProducts = data.results;
```
Comment (buty619) [inline src/views/GridFeatured.jsx]: same comment as above.

File: src/views/Home.jsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import HomePage from './HomePage';
+import Footer from '../components/Footer';
+import Header from '../components/Header';
+
+const Home = () => {
+  return (
+    <div className=''>
```
Comment (buty619) [inline src/views/Home.jsx]: please remove the className if not need it, also you can use a fragment here

File: src/views/HomePage.jsx [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+import bannersData from '../assets/mocks/en-us/featured-banners.json'
+import GridCategories from './GridCategories'
+import GridFeatured from './GridFeatured'
+import BannerSlider from '../components/BannerSlider'
+import styles from '../styles/HomePage.module.scss'
+
+const HomePage = () => {
+  return (
+    <div className={`d-flex flex-column p-3 p-md-4 gap-5 ${styles.homepage}`}>
+      <div>
```
Comment (buty619) [inline src/views/HomePage.jsx]: why you need to wrap each component inside a div?

**Review by buty619 — CHANGES_REQUESTED** [review]
great job¡¡ small comments

File: package.json (hunk 1 above)
Comment (johannrp27) [reply package.json:12]: Yes you're right, I'll change it

File: src/react-fontawesome-config.js (hunk 4 above)
Comment (johannrp27) [reply src/react-fontawesome-config.js:5]: It's a config file where you only import icons you'll use and make them available for whole application. However I would consider replacing it with a smaller library based how many icons I will need

**Pull request merged**